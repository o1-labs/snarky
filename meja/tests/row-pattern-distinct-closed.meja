let f = fun (x : [< `A | `B > `A]) => {
  switch (x) {
    | `C => ()
    | _ => ()
  };
};
