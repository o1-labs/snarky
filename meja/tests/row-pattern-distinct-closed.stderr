File "tests/row-pattern-distinct-closed.meja", line 3, characters 6-8:
Error: Incompatible types ([<`A | `B > `A] as 'a) and ([>`C] as 'b):
       Cannot unify ([<`A | `B > `A] as 'a) and ([>`C] as 'b):
       the constructor `C is not present in both.
